zeebe/protocol/src/main/java/io/camunda/zeebe/protocol/record/intent/UserTaskIntent.java

```diff
@@ -76,7 +76,13 @@ public enum UserTaskIntent implements ProcessInstanceRelatedIntent {
    * Note that the changes are only applied to the User Task instance after all Task Listeners have
    * been handled and none denied the operation.
    */
-  CORRECTED(18);
+  CORRECTED(18),
+
+  /**
+   * Represents the intent indicating that the User Task will not be assigned and the user task's
+   * assignee remains unchanged. The user task will be reverted to the CREATED lifecycle state.
+   */
+  ASSIGNMENT_DENIED(19);
```

Comment:
👍🏻 I like the decision to introduce a distinct intent for the denial of assignment operation. While it might have been possible to use a shared event for denied (complete/assign) operations, having separate intents improves clarity and makes the record stream easier to read and debug.